src/coreclr/jit/emit.h

```diff
@@ -780,6 +780,9 @@ class emitter
         unsigned _idCallRegPtr : 1; // IL indirect calls: addr in reg
         unsigned _idCallAddr : 1;   // IL indirect calls: can make a direct call to iiaAddr
         unsigned _idNoGC : 1;       // Some helpers don't get recorded in GC tables
+#if defined(TARGET_XARCH)
+        unsigned _idEvexbContext : 1; // does EVEX.b need to be set.
```

Comment:
Just some notes, we're going to need potentially up to 7 new bits: https://github.com/dotnet/runtime/pull/84821#discussion_r1174097945

We'll also maybe need bits to represent +1 register for the kmask scenario. `vfixupimm` and `vpternlog` are `2 registers` + `1 register or addressing mode` + `1 constant` (this is everything we support today). We then also have + `1 mask register` + a bit for `EVEX.b` + a bit for `EVEX.Z` + a bit for `EVEX.L'L` (all of which need to be carried until the `code` is constructed).

The `broadcast` case is already going to require some `addressing mode`. The same `EVEX.b` bit may also be used to flag the `rounding control` or `SAE control`, however, which doesn't require addressing. The `EVEX.Z` bit is only used with the `opmask` register but the `opmask` can be used independently of `addressing`.

It would be nice to squeeze these bits into the spare padding we have available, if possible, but its not "required". Doing so may help throughput due to having less `instrDesc*` kinds to handle.

Reply:
We should think more about this. Every bit that is put here takes away from "small" constants. We should also consider that code requiring EVEX information is likely to be extremely rare compared to non-EVEX instructions. For embedded broadcast, which requires an addressing mode, maybe `emitAddrMode iiaAddrMode` can carry the data, for example.

Reply:
Ah, I didn't realize these spare bits were used to hold the small constants 👍 

For clarification, what is currently defined as a "small" constant? On the x86 side we have:
* `imm8` - most commonly used, used extensively by hwintrinsics and general instructions alike
* `imm16` - extremely rare as emitting 16-bit operations is itself very rare and typically more expensive than a 32-bit operation
* `imm32` - second most common and used by calls, jumps, etc
* `imm64` - 64-bit only and only for 1 instruction

My naive guess would then be that we want to reserve `8-bits` for "small constants" and never shrink past that. 16/32-bit constants would then be considered non-small and part of the regular `instrDesc`. 64-bit constants would be "unique" since it impacts 1 instruction and is rare.

Reply:
Well, a "small constant" in the emitter is "whatever space is left over in the first 64 bits of the instrDesc". Basically, a "small" instrDesc is 64-bits. `_idSmallCns` takes up whatever space is left over after the instruction opcode, 2 registers, GC type, etc. It's 7-12 bits, currently, depending on architecture. If the value doesn't fit in that, we have to allocate an `instrDescCns` or some other `instrDesc` subclass to hold it. Adding more bits in the "small instrDesc" section pushes some constants to require a bigger `instrDesc` format.

Reply:
I'll take responsibility for splitting this into its own `instrDesc` as part of adding the support for embedded rounding control and masking support.